marklogic/documents.py

```diff
@@ -262,23 +263,30 @@ def _extract_values_from_header(part) -> dict:
     Returns a dict containing values about the document content or metadata.
     """
     encoding = part.encoding
-    disposition = part.headers["Content-Disposition".encode(encoding)].decode(encoding)
-    disposition_values = {}
-    for item in disposition.split(";"):
-        tokens = item.split("=")
-        # The first item will be "attachment" and can be ignored.
-        if len(tokens) == 2:
-            disposition_values[tokens[0].strip()] = tokens[1]
+    disposition = part.headers["Content-Disposition".encode(encoding)].decode(
+        encoding
+    )
 
     content_type = None
     if part.headers.get("Content-Type".encode(encoding)):
-        content_type = part.headers["Content-Type".encode(encoding)].decode(encoding)
+        content_type = part.headers["Content-Type".encode(encoding)].decode(
+            encoding
+        )
 
-    uri = disposition_values["filename"]
-    if uri.startswith('"'):
-        uri = uri[1:]
-    if uri.endswith('"'):
-        uri = uri[:-1]
+    content_disposition_header = part.headers[
+        "Content-Disposition".encode(encoding)
+    ].decode(encoding)
+    msg = Message()
+    msg["content-disposition"] = content_disposition_header
+    uri = msg.get_filename()
+    partial_disposition = disposition.replace("attachment;", "")
```

Comment:
This looks good. To make things a little simpler though - since we know we only care about "category" and "versionId", I think we should do the following:

1. No need to do this replacement of "attachment;", we only need to replace the URI as done on the next line.
2. Change the `if tokens[0] not in` line below to instead be `if tokens[0] in ["category", "versionId"]` . That will make it explicit as to what we care about from the disposition. 

I did a quick test of the following and all the tests passed:

```
    disposition_values = {}
    for item in disposition.replace(uri, "").split(";"):
        tokens = item.split("=")
        key = tokens[0].strip()
        if key in ["category", "versionId"]:
            disposition_values[key] = tokens[1]
```